tests/5.2/unified_shared_mem/test_target_OpertrOvrld.cpp

```diff
@@ -0,0 +1,93 @@
+//===--  test_target_OpertrOvrld.cpp-------------------------------------===//
+//
+// OpenMP API Version 5.2
+//
+// Description
+// This test case tests the working of the following aspects:
+// 1) Working of "requires unified_shared_memory"
+// 2) Testing operator overloading on gpu
+//===----------------------------------------------------------------------===//
+
+#include <iostream>
+#include <omp.h>
+#include "ompvv.h"
+
+using namespace std;
+
+#pragma omp requires unified_shared_memory
+class Fraction {
+  int num;
+  int denom;
+  public:
+    Fraction(int num, int denom) {
+      this->num = num;
+      this->denom = denom;
+    }
+
+  Fraction operator+(Fraction const &f1) {
+    int lcm = denom * f1.denom;
+    int x = lcm/denom;
+    int y = lcm/f1.denom;
+    int Rnum = x*num + y*f1.num;
+    int Rdenom = lcm;
+    int gcd = 1;
+    for (int i = 1; i <= Rnum; ++i) {
+      if (((Rnum % i) == 0) && ((Rdenom % i) == 0)) {
+        gcd = i;
+      }
+    }
+
+  Fraction New(Rnum/gcd, Rdenom/gcd);
+    return New;
+  }
+
+  Fraction operator*(Fraction const &f1) {
+    int Nnum = num * f1.num;
+    int Ndenom = denom * f1.denom;
+    int gcd = 1;
+    for (int i = 1; i <= Nnum; ++i) {
+      if (((Nnum % i) == 0) && ((Ndenom % i) == 0)) {
+         gcd = i;
+      }
+    }
+  Fraction New(Nnum/gcd, Ndenom/gcd);
+    return New;
+  }
+
+  pair<int, int> GetVals() {
+    pair<int, int> p(num, denom);
+    return p;
+  }
+};
+
+
+int main() {
+  OMPVV_TEST_OFFLOADING;
+  int errors = 0;
+  int *Errs = (int *)omp_target_alloc(sizeof(int), 0);
```

Comment:
Why do we need device ptr here ?
Prefer to use int Errs, then map it tofrom on the target directive.

Reply:
Used int Errs and map for tofrom...